modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportAnimalQTL.pm

```diff
@@ -0,0 +1,345 @@
+
+use strict;
+use warnings;
+
+use File::Path qw(make_path);
+use File::stat;
+use POSIX 'strftime';
+
+use base ('Bio::EnsEMBL::Variation::Pipeline::PhenotypeAnnotation::BasePhenotypeAnnotation');
+
+my %source_info;
+
+# AnimalQTLdb URL set up for Ensembl: https://www.animalgenome.org/QTLdb/export/ENS83H19HZS/
+#my $animalqtl_url = 'https://www.animalgenome.org/cgi-bin/QTLdb/index';
+my %animalQTL_species_url = (
+  gallus_gallus => 'https://www.animalgenome.org/QTLdb/export/ENS83H19HZS/QTL_GG_5.0.gff.txt.gz', #Gallus gallus
+  sus_scrofa => 'https://www.animalgenome.org/QTLdb/export/ENS83H19HZS/QTL_SS_11.1.gff.txt.gz', #Sus scrofa
+  ovis_aries => 'https://www.animalgenome.org/QTLdb/tmp/QTL_OAR_3.1.gff.txt.gz',  # Ovis aries #TODO: replace with the one in export once it is there
+  bos_taurus => 'https://www.animalgenome.org/QTLdb/export/ENS83H19HZS/QTL_BovARS_1.2.gff.txt.gz', #Bos taurus
+  equus_caballus => 'https://www.animalgenome.org/cgi-bin/QTLdb/EC/download?file=gbpEC_2.0', #Equus caballus
+);
+
+my %animalQTL_species_fileNames = (
+  gallus_gallus => 'gallus_gallus_gbp_5.0.gff3.gz', #Gallus gallus
+  sus_scrofa => 'sus_scroafa_gbp_11.1.gff3.gz', #Sus scrofa
+  ovis_aries => 'ovis_aries_gbp_3.1.gff3.gz',  # Ovis aries #TODO: replace with the one in export once it is there
+  bos_taurus => 'bos_taurus_gbp_1.2.gff3.gz', #Bos taurus
+  equus_caballus => 'equus_caballus_gbp_2.0.gff3.gz', #Equus caballus
+);
+
+my %animalQTL_species_ok = (
+  gallus_gallus => 0, #Gallus gallus same Ensembl assembly as AnimalQTL
+  sus_scrofa => 1, #Sus scrofa
+  ovis_aries => 1,  # Ovis aries
+  bos_taurus => 1, #Bos taurus
+  equus_caballus => 0, #Equus caballus
+);
+
+
+sub fetch_input {
+  #create output folder structure and fetches input files
+  my $self = shift;
+
+  my $pipeline_dir = $self->required_param('pipeline_dir');
+  my $species      = $self->required_param('species');
+
+  # import specific constants
+  %source_info = (source_description => 'The Animal Quantitative Trait Loci (QTL) database (Animal QTLdb) is designed to house all publicly available QTL and association data on livestock animal species',
+                  source_url => 'https://www.animalgenome.org/cgi-bin/QTLdb/index',
+                  object_type => 'QTL',
+                  #source_version  will be set based on file date
+                  source_status     => 'germline',
+                  source_name       => 'Animal_QTLdb', #source name in the variation db
+                  source_name_short => 'AnimalQTLdb',  #source identifier in the pipeline
+                  threshold => $self->required_param('threshold_qtl'),
+                  );
+
+  #create workdir folder
+  my $workdir = $pipeline_dir."/".$source_info{source_name_short}."/".$species;
+  make_path($workdir);
+  $self->workdir($workdir);
+
+  return unless $animalQTL_species_ok{$species};
+
+  $self->debug($self->param('debug_mode'));
+  $self->core_db_adaptor($self->get_species_adaptor('core'));
```

Comment:
Could we rewrite `core_db_adaptor` to call directly `get_species_adaptor('core')` if it hasn't been populated ?
Same for the other adaptors ?